src/eduid_userdb/user.py

```diff
@@ -229,6 +230,18 @@ def _parse_tous(self):
         _tou = self._data_in.pop('tou', [])
         self._tou = ToUList(_tou)
 
+    def _parse_locked_identity(self):
+        self._locked_identity = None
+        supported_identity_types = ['nin']
+        _locked_identity = self._data_in.pop('locked_identity', None)
+        if _locked_identity:
+            if _locked_identity.get('identity_type') not in supported_identity_types:
```

Comment:
I would avoid checking identity_type essentially twice by replacing this "if x not in supported_identity_types" with an

if _locked_identity == 'nin':
 ...
else
  raise UserDBValueError()